google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/RpcMethodMappings.java

```diff
@@ -0,0 +1,1942 @@
+    ObjectAcl.update(a);
+
+    Objects.delete(a);
+    Objects.get(a);
+    Objects.insert(a);
+    Objects.list(a);
+    Objects.patch(a);
+    Objects.update(a);
+    Objects.compose(a);
+    Objects.rewrite(a);
+    Objects.copy(a);
+
+    ServiceAccount.get(a);
+    ServiceAccount.put(a);
+
+    validateMappingDefinitions(failOnDuplicateIds, a);
+
+    funcMap = Multimaps.index(a, RpcMethodMapping::getMethod);
+    reportMappingSummary();
+  }
+
+  public Collection<RpcMethodMapping> get(RpcMethod key) {
+    return funcMap.get(key);
+  }
+
+  public Set<Integer> differenceMappingIds(Set<Integer> usedMappingIds) {
+    return Sets.difference(
+        funcMap.values().stream().map(RpcMethodMapping::getMappingId).collect(Collectors.toSet()),
+        usedMappingIds);
+  }
+
+  private void validateMappingDefinitions(
+      boolean failOnDuplicateIds, ArrayList<RpcMethodMapping> a) {
+    ListMultimap<Integer, RpcMethodMapping> idMappings =
+        MultimapBuilder.hashKeys()
+            .arrayListValues()
+            .build(Multimaps.index(a, RpcMethodMapping::getMappingId));
+    String duplicateIds =
+        idMappings.asMap().entrySet().stream()
+            .filter(e -> e.getValue().size() > 1)
+            .map(Entry::getKey)
+            .map(i -> Integer.toString(i))
+            .collect(Collectors.joining(", "));
+    if (!duplicateIds.isEmpty()) {
+      String message = "duplicate mapping ids present: [" + duplicateIds + "]";
+      if (failOnDuplicateIds) {
+        throw new IllegalStateException(message);
+      } else {
+        LOGGER.warning(message);
+      }
+    }
+  }
+
+  private void reportMappingSummary() {
+    int mappingCount = funcMap.values().stream().mapToInt(m -> 1).sum();
+    LOGGER.info("Current total number of mappings defined: " + mappingCount);
+    String counts =
+        funcMap.asMap().entrySet().stream()
+            .map(
+                e -> {
+                  RpcMethod rpcMethod = e.getKey();
+                  Collection<RpcMethodMapping> mapings = e.getValue();
+                  return String.format(
+                      "\t%s.%s: %d",
+                      rpcMethod
+                          .getClass()
+                          .getName()
+                          .replace("com.google.cloud.storage.conformance.retry.RpcMethod$", "")
+                          .replace("$", "."),
+                      rpcMethod,
+                      mapings.size());
+                })
+            .sorted()
+            .collect(Collectors.joining("\n", "\n", ""));
+    LOGGER.info("Current number of mappings per rpc method: " + counts);
+    OptionalInt max =
+        funcMap.values().stream().map(RpcMethodMapping::getMappingId).mapToInt(i -> i).max();
+    if (max.isPresent()) {
+      LOGGER.info(String.format("Current max mapping index is: %d%n", max.getAsInt()));
+    } else {
+      throw new IllegalStateException("No mappings defined");
+    }
+  }
+
+  static final class Mappings {
```

Comment:
Soooo many mappings.

Reply:
Yeah, and the code formatter doesn't really help with condensing at all 😩